---
- hosts: localhost
  gather_facts: no

  tasks:
  - name: Handle the error
    block:
      - name: Run ansible-lint
        command:
          cmd: ansible-lint --exclude=ci --force-color
          chdir: "{{ lookup('env', 'CI_PROJECT_DIR') }}"
        register: result
        ignore_errors: true
      
      - name: print result to file
        shell: echo "{{result.stdout[:3000]}}" > /tmp/result.txt

      - name: cat result
        shell: cat /tmp/result.txt

      - name: Send fail var to telegram task
        include_tasks: send_telegram.yml
        vars:
          event_name: 'finished'
          error_code: "{{ lookup('ansible.builtin.file','/tmp/result.txt') }}"
        when: "'0 failure(s)' in result.stderr"

      - name: Send finish var to telegram task
        include_tasks: send_telegram.yml
        vars:
          event_name: 'failed'
          error_code: "{{ lookup('ansible.builtin.file','/tmp/result.txt') }}"
        when: "not '0 failure(s)' in result.stderr"
    rescue: 
      - name: Send message that error occured while execute CI
        include_tasks: send_telegram.yml
        vars:
          event_name: 'Failed occured while executing Pipeline'
          error_code: "Error occured in pipeline"